docs/src/userguide/numpy_tutorial.rst

```diff
@@ -70,31 +64,36 @@ However there are several options to automate these steps:
 
 1. The `SAGE <http://sagemath.org>`_ mathematics software system provides
    excellent support for using Cython and NumPy from an interactive command
-   line (like IPython) or through a notebook interface (like
+   line or through a notebook interface (like
    Maple/Mathematica). See `this documentation
-   <http://www.sagemath.org/doc/prog/node40.html>`_.
-2. A version of `pyximport <http://www.prescod.net/pyximport/>`_ is shipped
-   with Cython, so that you can import pyx-files dynamically into Python and
+   <http://doc.sagemath.org/html/en/developer/coding_in_cython.html>`_.
+2. Cython can be used as an extension within a Jupyter notebook,
+   making it easy to compile and use Cython code with just a ``%%cython``
+   at the top of a cell. For more information see
+   :ref:`Using the Jupyter Notebook <jupyter-notebook>`.
+3. A version of pyximport is shipped with Cython,
+   so that you can import pyx-files dynamically into Python and
    have them compiled automatically (See :ref:`pyximport`).
-3. Cython supports distutils so that you can very easily create build scripts
+4. Cython supports distutils so that you can very easily create build scripts
    which automate the process, this is the preferred method for full programs.
-4. Manual compilation (see below)
+   See :ref:`Compiling with distutils <compiling-distutils>`.
+5. Manual compilation (see below)
 
 .. Note::
     If using another interactive command line environment than SAGE, like
-    IPython or Python itself, it is important that you restart the process
+    IPython, Jupyter Notebook or Python itself, it is important that you restart the process
```

Comment:
This doesn't really apply directly to Jupyter, where users are not expected to "compile" anything but things are happening behind their back when running a Cython cell. When you change something in a Jupyter Cython cell, it will automatically be recompiled. If you don't change anything, it won't be.